refactor(client-presence): Refactor/Rename APIs (#24384)
microsoft/FluidFramework: .changeset/sour-mirrors-wave.md

> ---
> "@fluidframework/presence": minor
> ---
> ---
> "section": other
> ---

Reviewer: ```suggestion
---
"@fluidframework/presence": minor
"__section": other
---
```

Changeset format has changed recently, so we need to do a little manual fix up here.

> Presence API renames

Reviewer: ```suggestion
Presence APIs have been renamed
```

The following API changes have been made to improve clarity and consistency:

> | Original | New |

Reviewer: ```suggestion
| Before 2.33.0 | 2.33.0 |
```

| `LatestMapValueManager.clientValue` | `LatestMap.getRemote` |
| `LatestMapValueManager.clientValues` | `LatestMap.getRemotes` |
| `LatestMapValueManagerEvents` | `LatestMapEvents` |
| `LatestValueClientData` | `LatestClientData` |
| `LatestValueData` | `LatestData` |
| `LatestValueManager` | `Latest` |
| `LatestValueManager.clients` | `Latest.getStateAttendees` |
| `LatestValueManager.clientValue` | `Latest.getRemote` |
| `LatestValueManager.clientValues` | `Latest.getRemotes` |
| `LatestValueManagerEvents` | `LatestEvents` |
| `LatestValueMetadata` | `LatestMetadata` |
| `PresenceEvents.attendeeDisconnected` | `AttendeesEvents.attendeeDisconnected`|
| `PresenceEvents.attendeeJoined` | `AttendeesEvents.attendeeConnected`|
| `PresenceNotifications` | `NotificationsWorkspace` |
| `PresenceNotificationsSchema` | `NotificationsWorkspaceSchema` |
| `PresenceStates` | `StatesWorkspace` |
| `PresenceStatesEntries` | `StatesWorkspaceEntries` |
| `PresenceStatesSchema` | `StatesWorkspaceSchema` |
| `PresenceWorkspaceAddress` | `WorkspaceAddress` |
| `PresenceWorkspaceEntry` | `StatesWorkspaceEntry` |
| `SessionClientStatus` | `AttendeeStatus` |
| `ValueMap` | `StateMap` |

> Note: To fully replace OLD `Latest` and `LatestMap` functions, you should import `StateFactory` and call `StateFactory.latest` and `StateFactory.latestMap` respectively. NEW `Latest` and `LatestMap` APIs replace `LatestValueManager` and `LatestMapValueManager`.

Reviewer: ```suggestion
> [!NOTE]
> To fully replace the former `Latest` and `LatestMap` functions, you should import `StateFactory` and call `StateFactory.latest` and `StateFactory.latestMap` respectively. The new `Latest` and `LatestMap` APIs replace `LatestValueManager` and `LatestMapValueManager` respectively.
```
